iox_data_generator/Cargo.toml

```
@@ -0,0 +1,37 @@
+[package]
+name = "data_generator"
```

It's somewhat expected that the crate directory and package name match, so I would make this:

```suggestion
name = "iox_data_generator"
```